cabal-install/src/Distribution/Client/Manpage.hs

```diff
@@ -50,24 +55,44 @@ manpageCmd pname commands flags
     | fromFlagOrDefault False (manpageRaw flags)
     = putStrLn contents
     | otherwise
-    = do
-        let cmd  = "man"
-            args = ["-l", "-"]
-
-        (mb_in, _, _, ph) <- createProcess (Process.proc cmd args)
-            { Process.std_in  = Process.CreatePipe
-            , Process.std_out = Process.Inherit
-            , Process.std_err = Process.Inherit
-            }
-
-        -- put contents
-        for_ mb_in $ \hin -> do
-            hPutStr hin contents
-            hClose hin
-
-        -- wait for process to exit, propagate exit code
-        ec <- Process.waitForProcess ph
-        exitWith ec
+    = ignoreSigPipe $ do
+        -- 2021-10-08, issue #7714
+        -- @cabal man --raw | man -l -@ does not work on macOS/BSD,
+        -- because BSD-man does not support option @-l@, rather would
+        -- accept directly a file argument, e.g. @man /dev/stdin@.
+        -- The following works both on macOS and Linux
+        -- (but not on Windows out-of-the-box):
+        --
+        --   cabal man --raw | nroff -man /dev/stdin | less
+        --
+        -- So let us simulate this!
+
+        -- Feed contents into @nroff -man /dev/stdin@
+        (formatted, _errors, ec1) <- rawSystemStdInOut
+          Verbosity.normal
+          "nroff"
+          [ "-man", "/dev/stdin" ]
+          Nothing  -- Inherit working directory
+          Nothing  -- Inherit environment
+          (Just $ IODataText contents)
+          IODataModeText
+
+        unless (ec1 == ExitSuccess) $ exitWith ec1
+
+        -- Pipe output of @nroff@ into @less@
+        (Just inLess, _, _, procLess) <- createProcessWithEnv
+          Verbosity.normal
+          "less"
```

Comment:
`most` is much better and also seems to have no problem with the newlines I think.

One way would be to check the `$PAGER` env var and fall back to `less` if that isn't set.

Reply:
https://github.com/andreasabel/cabal/pull/1

Reply:
using `less` is borked here:

## Less

![Screenshot_2022-01-20_11-51-49](https://user-images.githubusercontent.com/1241845/150325056-33474721-da53-4ffd-8194-31fa8ae79a91.png)

## Most

![Screenshot_2022-01-20_11-52-06](https://user-images.githubusercontent.com/1241845/150325071-f567fd27-4dea-438b-add9-30836e5c197a.png)

Reply:
less needs `-R` (or `-r`) to display colours

Reply:
> less needs `-R` (or `-r`) to display colours

Is that safe to add unconditionally?

```
       -r or --raw-control-chars
              Causes "raw" control characters to be displayed.  The default is to display control characters using the caret notation; for  ex‐
              ample,  a  control-A (octal 001) is displayed as "^A".  Warning: when the -r option is used, less cannot keep track of the actual
              appearance of the screen (since this depends on how the screen responds to each type of control character).  Thus,  various  dis‐
              play problems may result, such as long lines being split in the wrong place.

              USE OF THE -r OPTION IS NOT RECOMMENDED.
```

Reply:
Seems `-R` should be ok and also works under FreeBSD

Reply:
https://github.com/andreasabel/cabal/pull/1/commits/c4140b19e827a97fc9152dc60fbe763ce5d02529